### DISCUSSION TITLE
Is it possible to drop the red dot before the line that has error, change the line number to red?

### DISCUSSION THREAD

**Discussion by stokhos** [helix-editor/helix#10367] [Ideas]
Is it possible to drop the red dot before the line that has an error, and change the line number to red?
Or maybe add an option for this.  Doing this will save some spaces

Best

**Comment by grv07** [comment]
Since we are using the same space for breakpoint and diagnostic. I have the following questions.

1. What should be the style of the line number if it has a breakpoint and diagnostic?
Current behavior: It will show the breakpoint if available or show the diagnostic.

2. Since we have diff styles/colors for the current line. What should be the style in case of if the line has both the cursor and the diagnostic/breakpoint? 
Current behavior: Line number color and diagnostic/breakpoint are in separate spaces so no problem.

**Comment by stokhos** [comment]
> Since we are using the same space for breakpoint and diagnostic. I have the following questions.
> 
>     1. What should be the style of the line number if it has a breakpoint and diagnostic?
>        Current behavior: It will show the breakpoint if available or show the diagnostic.

| we could change the line number alignment. Line numbers are right-aligned; we could change the line number to a red left-aligned number.

>     2. Since we have diff styles/colors for the current line. What should be the style in case of if the line has both the cursor and the diagnostic/breakpoint?
>        Current behavior: Line number color and diagnostic/breakpoint are in separate spaces so no problem.
| No changes in current line styles if there is no error or breakpoint in this line.

@kirawi I'd like to work on this. Could you tell me where I should get started?

**Comment by kirawi** [comment]
You could take a look at https://github.com/helix-editor/helix/blob/65edf9c19880fe4fd068db6d86c4c2830c8a91f6/helix-term/src/ui/editor.rs#L710-L826

**Comment by aral** [comment]
A couple of quick accessibility/design notes:

- Colour should never be the only differentiator (think: colour blindness)
- If this is going to be implemented, it should be a non-default preference setting.
- Preferences shouldn’t be seen as an easy alternative to making design decisions.

hth :)
